syntax = "proto3";

package remote_target;

service RemoteTarget {
	// Request a connection
	rpc RequestConnection (EmptyReq) returns (Ping) {}
	// Confirmes the connection
	rpc ConfirmConnection (Pong) returns (ConnectionConfirmed) {}
	// Terminates the connection
	rpc Disconnect (ConnectionID) returns (ConnectionTerminated) {}
	// Available only after confirmed connection
	rpc GetDeviceSpec (ConnectionID) returns (DeviceSpec) {}
}

message EmptyReq {
}

message ConnectionID {
	int32 connectionID = 1;
}

message ConnectionConfirmed {
	bool success = 1;
	int32 connectionID = 2;
}

message ConnectionTerminated {
	int32 tod = 1;
}

message Ping {
	int32 connectionID = 1;
}

message Pong {
	int32 connectionID = 1;
}

message DeviceSpec {
	string name = 1;
	string OS_version = 2;
	string serial_number = 3;
	string comment = 4;
}
